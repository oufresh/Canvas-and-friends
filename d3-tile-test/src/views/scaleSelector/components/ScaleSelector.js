//@flow
import React from "react";
import styled from "styled-components";
import { branch, renderNothing } from "recompose";

const CELL_WIDTH = 25;
const CELL_HEIGHT = 24;

const ScaleArray = styled.div`
  font-family: Helvetica Neue, Helvetica, Arial, Liberation Sans, sans-serif;
  display: flex;
  flex-direction: row;
  background-color: transparent;
  text-align: center;
  user-select: none;
  height: 24px;
`;
const Scale = styled.div`
  width: ${CELL_WIDTH}px;
  height: ${CELL_HEIGHT}px;
  background-color: ${props =>
    props.current === true ? "rgba(255,40,40,0.60)" : "rgba(50,50,50,0.5);"};
  color: ${props => (props.current ? "cyan" : "#F0F0F0")};
  font-weight: ${props => (props.current ? "bold;" : "")};
  padding-top: 4px;

  &:hover {
    cursor: pointer;
  }
`;

type Props = {|
  onClick: (string, number) => any,
  scales?: Array<number>,
  selectedScale: number,
  visible: boolean,
  uuid: string
|};

const ScaleSelector = ({
  onClick,
  scales = [],
  selectedScale,
  uuid
}: Props) => {
  const w = scales.length * CELL_WIDTH + 2;
  return (
    <ScaleArray style={{ width: w + "px" }}>
      {scales.map(item => (
        <Scale
          key={`scale-${item}`}
          current={selectedScale === item}
          onClick={() => onClick(uuid, item)}
        >
          {item}
        </Scale>
      ))}
    </ScaleArray>
  );
};

// $FlowFixMe
const ScaleSelectorHOC = branch(
  (props: Props): boolean => !props.visible,
  renderNothing
)(ScaleSelector);

export { ScaleSelectorHOC };
