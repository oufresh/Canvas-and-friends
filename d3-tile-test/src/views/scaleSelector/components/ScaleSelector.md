### ScaleSelector Topologico MT

```js
    <ScaleSelector uuid="prova1" selectedScale={2} scales={[0,1,2,3,4]} visible={true}/>
```

### ScaleSelector CP

```js
    <ScaleSelector uuid="prova1" selectedScale={0} scales={[0,1]} visible={true}/>
```

### ScaleSelector hidden

```js
    <ScaleSelector uuid="prova1" selectedScale={0} scales={[0,1]} visible={false}/>
```
