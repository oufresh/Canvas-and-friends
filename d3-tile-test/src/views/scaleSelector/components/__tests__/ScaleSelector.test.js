import React from "react";
import { shallow } from "enzyme";
import { ScaleSelectorHOC } from "../ScaleSelector";

describe("ScaleSelector Component", () => {
  it("renders without a crash ", () => {
    const component = shallow(
      <ScaleSelectorHOC
        selectedScale={2}
        scales={[0, 1, 2, 3, 4]}
        visible={true}
        uuid="prova1"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
