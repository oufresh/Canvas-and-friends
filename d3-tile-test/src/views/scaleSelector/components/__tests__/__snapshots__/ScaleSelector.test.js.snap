// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScaleSelector Component renders without a crash  1`] = `
<ScaleSelector
  scales={
    Array [
      0,
      1,
      2,
      3,
      4,
    ]
  }
  selectedScale={2}
  uuid="prova1"
  visible={true}
/>
`;
